src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -1448,25 +1342,17 @@ subroutine ocn_compute_KPP_input_fields(statePool, forcingPool, meshPool, diagno
       call mpas_pool_get_array(tracersSurfaceFluxPool, 'activeTracersSurfaceFlux', activeTracersSurfaceFlux)
       call mpas_pool_get_array(tracersSurfaceFluxPool, 'activeTracersSurfaceFluxRunoff', activeTracersSurfaceFluxRunoff)
       ! allocate scratch space displaced density computation
-      call mpas_pool_get_field(scratchPool, 'densitySurfaceDisplaced', densitySurfaceDisplacedField)
       call mpas_pool_get_field(scratchPool, 'thermalExpansionCoeff', thermalExpansionCoeffField)
       call mpas_pool_get_field(scratchPool, 'salineContractionCoeff', salineContractionCoeffField)
-      call mpas_allocate_scratch_field(densitySurfaceDisplacedField, .true., .false.)
       call mpas_allocate_scratch_field(thermalExpansionCoeffField, .true., .false.)
       call mpas_allocate_scratch_field(salineContractionCoeffField, .true., .false.)
 
-      densitySurfaceDisplaced => densitySurfaceDisplacedField % array
       thermalExpansionCoeff => thermalExpansionCoeffField % array
       salineContractionCoeff => salineContractionCoeffField % array
 
       ! Only need to compute over the 0 and 1 halos
       nCells = nCellsArray( 3 )
 
-      ! compute EOS by displacing SST/SSS to every vertical layer in column
-      call ocn_equation_of_state_density(statePool, diagnosticsPool, meshPool, scratchPool, nCells, 0, 'surfaceDisplaced', &
-                                         densitySurfaceDisplaced, err, thermalExpansionCoeff, salineContractionCoeff, &
```

Comment:
@vanroekel This EOS call was removed, but thermalExpansionCoeff and salineContractionCoeff are used below. Was that an oversight? I added it back in with `'absolute', density` but global runs still differ from ocean/develop.

Reply:
yes, this was overzealous removal of ```densitySurfaceDisplaced```. and it is not surprising after looking at this more that the PR is non bfb. we are changing the alpha and beta values used in KPP.  My preferred long term solution is to change this call to compute potentialDensity and return thermalExpansionCoeff and salineContractionCoeff associated with that displacement.  I expect that change to be non-BFB but is unlikely to be climate changing.  I would suggest we hold off on this PR until I can fully test a change to potential density.